pxtcompiler/emitter/decompiler.ts

```diff
@@ -1474,13 +1474,20 @@ ${output}</xml>`;
                 r.inputs = [getDraggableVariableBlock("VAR", renamed)];
 
                 if (condition.operatorToken.kind === SK.LessThanToken) {
-                    const ex = mkExpr("math_arithmetic");
-                    ex.fields = [getField("OP", "MINUS")];
-                    ex.inputs = [
-                        getValue("A", condition.right, numberType),
-                        getValue("B", 1, numberType)
-                    ];
-                    r.inputs.push(mkValue("TO", ex, wholeNumberType));
+                    const unwrappedRightSide = unwrapNode(condition.right);
+                    if (unwrappedRightSide.kind === SK.NumericLiteral) {
+                        const decrementedValue = parseFloat(unwrappedRightSide.getFullText()) - 1;
```

Comment:
I've only ever used ``getFullText`` when debugging other than this - is there a helper or better way to handle this sort of thing that I'm forgetting?

Reply:
Just use `unwrappedRightSide.text` (you'll need to cast to a `LiteralExpression). I think `getFullText()` will include comment trivia.